<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ASD Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="meeting.html">Meeting</a>
</li>
<li>
  <a href="Note.html">Note</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<h1 style="color:grey;font-family:arial; position:relative; top:15px;text-align:center;">
TADA-A-plus<img src="website_materials/research_website_logo.png"  style="float:right;width:200px;height:150px;">
</h1>
<p><br></p>
<hr />
<p><br></p>
<div id="analyze-various-coding-annotations" class="section level1">
<h1>1. Analyze various coding annotations</h1>
<p><br></p>
<blockquote>
<p>Annotations include</p>
</blockquote>
<ul>
<li>Ribosnitch (allele-specific)</li>
<li>RBP binding sites</li>
<li>conservation (GERP)</li>
<li>Polyphen-damaging (Mis3)</li>
<li>Loss-of-function and Gain-of-function</li>
<li>CADD</li>
</ul>
<div id="for-wgs-data" class="section level2">
<h2>1.1 For WGS data</h2>
<p>Use manuscript data/MSSNG/Simons_519, scaling factors are computed using WGS data, based on coding regions and regions 10kb within TSSs.</p>
<div id="manuscript-meta-analysis-data" class="section level3">
<h3>1.1.1 manuscript meta-analysis data</h3>
<ul>
<li>coding RR estimation.<a href="analysis/180214_results_summary_manuscript_data_RR_estimation_with_mutrate_adjusted_for_div_coding_features.html">link</a>(Mutation rate adjusted using all mutations of manuscript windows including exons, promoters and promoter-proximal non-coding regions )</li>
</ul>
</div>
<div id="mssng-data" class="section level3">
<h3>1.1.2 MSSNG data</h3>
<ul>
<li><p>coding RR estimation (scaling file doesn’t account for div score).<a href="analysis/180216_RR_estimate_DATA_Yuen_2017_ANNO_RR_estimate.html">link</a>(Mutation rate adjusted using all mutations of manuscript windows including exons, promoters and promoter-proximal non-coding regions )</p></li>
<li>coding RR estimation (scaling file accounting for div score).<a href="analysis/180216_with_div_adjusted_RR_estimate_DATA_Yuen_2017_ANNO_RR_estimate.html">link</a></li>
<li><p><a href="analysis/180215_recalibrate_mutation_rate_with_div_score_for_MSSNG_and_Simons_519.html">Adjusting mutation rate accounting for div score</a> (Mutation rate adjusted using all mutations of manuscript windows including exons, promoters and promoter-proximal non-coding regions )</p></li>
<li><p>About 75% of the Yuen et al. data used in the TADA-A manuscript are also found in the denovo-db Yuen data.</p></li>
</ul>
<p>Execution code to do data partition and reading-in is <code>code/pipeline00_generate_R_code_for_estimating_RR_using_partitioned_data</code></p>
<ul>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model <a href="analysis/180412_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Yuen_2017_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR__with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR__with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR__with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Yuen_2017_cases_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="simons-data" class="section level3">
<h3>1.1.3 Simons data</h3>
<ul>
<li><p><a href="analysis/180215_recalibrate_mutation_rate_with_div_score_for_MSSNG_and_Simons_519.html">Adjusting mutation rate accounting for div score</a> (Mutation rate adjusted using all mutations of manuscript windows including exons, promoters and promoter-proximal non-coding regions )</p></li>
<li><p>coding RR estimation (scaling file accounting for div score).<a href="analysis/180216_with_div_adjusted_RR_estimate_Simons_519_new_version_ANNO_RR_estimate.html">link</a></p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model <a href="analysis/180412_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Simons_519_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Simons_519_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Simons_519_cases_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_cases_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="wong-nc-et-al.-control-data" class="section level3">
<h3>1.1.4 Wong NC et al. control data</h3>
<ul>
<li><p>coding RR estimation (scaling factors were calculated accouting for div score, using coding sequences and promoter-proximal non-coding sequences, used in the manuscript). <a href="analysis/180221_with_div_adjusted_RR_estimate_Wong_control_ANNO_RR_estimate.html">link</a> (Mutation rate adjusted using all mutations of manuscript windows including exons, promoters and promoter-proximal non-coding regions )</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model (The file title is a bit misleading as it contains “joint_model”, thought mutation rate adjustment were performed separately here) <a href="analysis/180412_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Wong_NC_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model (The file title is a bit misleading as it contains “joint_model”, thought mutation rate adjustment were performed separately here). Use informative prior <a href="./analysis/180425_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Wong_NC_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="netherland-258-control" class="section level3">
<h3>1.1.5 Netherland 258 control</h3>
<ul>
<li><p>coding RR estimation, mutation adjustment using all genes in the manuscript windows, including coding sequences, promoters and promoter-proximal noncoding regions, 3-mer mutation rate model <a href="analysis/180221_with_div_adjusted_RR_estimate_Neitherland_control_ANNO_RR_estimate.html">coding RR estimation accouting for div score</a></p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using nonTADA genes in coding regions to adjust mutation rates for the coding part <a href="analysis/180315_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Neitherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
</ul>
<p><a href="analysis/180405_recalibrate_mutation_rate_with_div_score_for_WGS_data_coding_part_using_nonfunctional_synonymous_ERV_mutrate.html">Adjust mutation rates using the coding parts, nonfunctional synonymous mutations, 7-mer ERV mutation rates model</a></p>
<p><a href="analysis/180405_recalibrate_mutation_rate_with_div_score_for_WGS_data_coding_part_using_nonfunctional_synonymous_ERV_mutrate_additional_covariates.html">Adjust mutation rates using the coding parts, nonfunctional synonymous mutations, 7-mer ERV mutation rates model, with additional mutation rate-adjusting covariates</a></p>
<ul>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using nonfunctional syn mutations to adjust mutation rates for the coding part <a href="analysis/180315_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Neitherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model <a href="analysis/180412_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Netherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Netherland_258_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="iceland-1549-control-control" class="section level3">
<h3>1.1.5 Iceland 1549 control control</h3>
<ul>
<li><p>coding RR estimation, mutation adjustment using all genes in the manuscript windows, including coding sequences, promoters and promoter-proximal noncoding regions, 3-mer mutation rate model <a href="analysis/180221_with_div_adjusted_RR_estimate_Iceland_control_ANNO_RR_estimate.html">coding RR estimation accouting for div score</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate model, using nonTADA genes to adjust mutation rates for the coding part <a href="analysis/180315_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iceland_1594_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using nonfunctional syn mutations to adjust mutation rates for the coding part <a href="analysis/180315_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iceland_1594_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model (The file title is a bit misleading as it contains “joint_model”, thought mutation rate adjustment were performed separately here) <a href="analysis/180412_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iceland_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model (The file title is a bit misleading as it contains “joint_model”, thought mutation rate adjustment were performed separately here). Use informative prior <a href="./analysis/180425_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iceland_control_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="simons-519-wgs-control" class="section level3">
<h3>1.1.6 Simons 519 WGS control</h3>
<ul>
<li><p><a href="/analysis/180417_recalibrate_mutation_rate_with_div_score_for_WGS_Simons_control_data_coding_part_using_nonfunctional_synonymous_ERV_mutrate_additional_covariates_noXY.html">Recalibrate mutation rates</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using nonfunctional syn mutations to adjust mutation rates for the coding part <a href="analysis/180417_with_div_mutrate_calibration_adjusted_RR_estimate_nonfunctional_syn_ERV_mutrate_additional_covariates_noXY_DATA_Simons_new_control_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model (The file title is a bit misleading as it contains “joint_model”, thought mutation rate adjustment were performed separately here) <a href="analysis/180417_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_noXY_DATA_Simons_new_control_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180502_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_noXY_DATA_Simons_new_control_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_noXY_DATA_Simons_new_control_cases_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_controls_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_519_controls_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="simons-1902-wgs-cases" class="section level3">
<h3>1.1.7 Simons 1902 WGS cases</h3>
<ul>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior, account for functional annotation in mutaion rate adjustment <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using uniform prior <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior, account for functional annotation in mutaion rate adjustment <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using informative prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Account for functional annotations in mutation rate adjustment. Using informative prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_for_footprinting_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, use uniform prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_using_uniform_prior.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_for_footprinting.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_for_footprinting_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_Simons_1902_WGS_cases_ANNO_RR_estimate_for_footprinting_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, use uniform prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_cases_ANNO_RR_estimate_for_footprinting_using_uniform_prior.html">link</a>.</p></li>
</ul>
</div>
<div id="siomons-1902-wgs-controls" class="section level3">
<h3>1.1.8 Siomons 1902 WGS controls</h3>
<ul>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior, account for functional annotation in mutaion rate adjustment <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using uniform prior <a href="analysis/180508_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>coding RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using LoF prior, account for functional annotation in mutaion rate adjustment <a href="analysis/180509_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>UTR RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior. <a href="analysis/180509_UTR_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using informative prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Account for functional annotations in mutation rate adjustment. Using informative prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, Using uniform prior <a href="analysis/180510_promoter_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_used_for_uniform_prior.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using informative priors from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>Intronic_spidex (windows contain spidex regions within 10kb of TSSs (non-promoter, non-UTR and non-coding) and deeper intronic spidex regions) RR estimation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180516_intronic_spidex_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, use uniform prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_using_uniform_prior.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, using informative prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_for_footprinting.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using WES prior from WES data. <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_for_footprinting_account_for_functional_anno.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, after the 1st round of mutation rate adjustment, also adjust for the effects of the functional annotations on mutation rate, using LoF prior from WES data. <a href="analysis/180520_Simons_1902_WGS_controls_ANNO_RR_estimate_for_footprinting_account_for_functional_anno_using_LoF_prior.html">link</a>.</p></li>
<li><p>5kb Promoter for footprinting RR estmation accouting for div score, using the 7-mer ERV mutation rate mode, using all mutations of coding, promoters, promoter-proximal noncoding regions and UTRs to to adjust mutation rates. Each study has its own separate mutation rate adjusting model, use uniform prior <a href="analysis/180520_promoter_5kb_with_div_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Simons_1902_WGS_controls_ANNO_RR_estimate_for_footprinting_using_uniform_prior.html">link</a>.</p></li>
</ul>
</div>
</div>
<div id="for-wes-data" class="section level2">
<h2>1.2 For WES data</h2>
<p>From denovo-db</p>
<table>
<thead>
<tr class="header">
<th>Name</th>
<th>PMID</th>
<th>Probands</th>
<th>Controls</th>
<th>Type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Iossifov</td>
<td>25363768</td>
<td>2508</td>
<td>1911</td>
<td>exome</td>
</tr>
<tr class="even">
<td>Krumm</td>
<td>25961944</td>
<td>2377</td>
<td>1786</td>
<td>exome</td>
</tr>
<tr class="odd">
<td>DeRubeis2014</td>
<td>25363760</td>
<td>2270</td>
<td>0</td>
<td>exome</td>
</tr>
</tbody>
</table>
<ul>
<li><p>Need to use the nonfunctional syn mutations to adjust mutation rates.</p></li>
<li><p>When adjusting mutation rates, need to generate synonymous mutation rates for each 50 bp window, the file is here <code>../data/Example_windows_coding_with_nonfunctional_syn_mutrate.bed</code></p></li>
<li><p><a href="analysis/180215_invesigate_ASD_DNM_in_denovo_db_for_WES_studies.html">clean dataset</a></p></li>
<li><p><a href="analysis/180215_recalibrate_mutation_rate_with_div_score_for_WES_data.html">Adjust mutation rate using nonfunctional synonymous mutations</a></p></li>
<li><p><a href="analysis/18308_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes.html">Ajust mutation rate using all the mutations of nonTADA genes, which are genes with coding prior smaller than 0.05</a></p></li>
<li><p><a href="analysis/180313_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes_ERV_mutrate.html">Ajust mutation rate using all mutations of nonTADA genes, but using the 7-mer ERV muation rate model</a></p></li>
<li><p><a href="analysis/180315_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonfunctional_syn_ERV_mutrate.html">Ajust mutation rate using nonfunctional synonymous mutations, using the 7-mer ERV mutation rate</a></p></li>
<li><p><a href="analysis/180405_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonfunctional_syn_ERV_mutrate_with_additional_covariates.html">Ajust mutation rate using nonfunctional synonymous mutations, using the 7-mer ERV mutation rate, additional rate-adjusting covariates</a></p></li>
<li><p><a href="analysis/180405_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonfunctional_syn_ERV_mutrate_with_additional_covariates_joint_model.html">Ajust mutation rate using nonfunctional synonymous mutations, using the 7-mer ERV mutation rate, additional rate-adjusting covariates, jointly model with data-specific covariates</a></p></li>
<li><p><a href="analysis/180410_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonfunctional_syn_ERV_mutrate_with_additional_covariates_noXY_joint_model.html">Adjust mutation rate using nonfunctional synonymous mutations, using the 7-mer ERV mutation rate, additional rate-adjusting covariates, jointly model with data-specific covariates, without using Krumm data, without X or Y chromosomes</a></p></li>
</ul>
<blockquote>
<p>It turns out that Krumm data collects mutations that were not previously discovered by Iossifov (for some major part of SSC data analyzed by Iossifov et al.)</p>
</blockquote>
<blockquote>
<p>The mr-eel pipeline has not been built to retireve mutation rates well on ChrX/Y. In practice, some of the ChrX/Y regions would be recovered but not all of them would.</p>
</blockquote>
<div id="iossifov-data-cases" class="section level3">
<h3>1.2.1 Iossifov data, cases</h3>
<ul>
<li><p>2767 coding mutations after filtering, 490 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations.<a href="analysis/180221_with_div_adjusted_RR_estimate_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in nonTADA genes.<a href="analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates <a href="analysis/180410_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using informative prior <a href="analysis/180425_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using informative prior, account for functional annotation in mutation rate adjustment <a href="analysis/180425_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iossifov_ANNO_RR_estimate_account_for_functional_anno.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using LoF prior, account for functional annotation in mutation rate adjustment <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iossifov_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a></p></li>
</ul>
</div>
<div id="iossifov-data-controls" class="section level3">
<h3>1.2.2 Iossifov data, controls</h3>
<ul>
<li><p>Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations.<a href="analysis/180221_with_div_adjusted_RR_estimate_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using nonTADA genes.<a href="analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.<a href="analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations, 7-mer ERV mutation rate model.<a href="analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates <a href="analysis/180410_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using informative prior <a href="analysis/180425_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using informative prior, account for functional annotation in mutation rate adjustment <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_Iossifov_control_ANNO_RR_estimate_account_for_functional_anno.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using LoF prior, account for functional annotation in mutation rate adjustment <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_Iossifov_control_ANNO_RR_estimate_account_for_functional_anno_used_for_LoF_prior.html">link</a></p></li>
</ul>
</div>
<div id="krumm-data-cases" class="section level3">
<h3>1.2.3 Krumm data, cases,</h3>
<ul>
<li><p>915 coding mutations after filtering, 119 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations.<a href="analysis/180221_with_div_adjusted_RR_estimate_Krumm_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in nonTADA genes.<a href="analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Krumm_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_DATA_Krumm_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_DATA_Iossifov_ANNO_RR_estimate.html">link</a></p></li>
</ul>
</div>
<div id="krumm-data-controls" class="section level3">
<h3>1.2.4 Krumm data, controls</h3>
<ul>
<li><p>Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations.<a href="analysis/180221_with_div_adjusted_RR_estimate_Krumm_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using all mutations in nonTADA genes.<a href="analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Krumm_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.<a href="analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.<a href="analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_control_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_DATA_Krumm_control_ANNO_RR_estimate.html">link</a></p></li>
</ul>
</div>
<div id="derubeis2014-data-cases" class="section level3">
<h3>1.2.5 DeRubeis2014 data, cases</h3>
<ul>
<li><p>1590 coding mutations after filtering, 315 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations. <a href="analysis/180221_with_div_adjusted_RR_estimate_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in using nonTADA genes. <a href="analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. <a href="analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_DATA_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific <a href="analysis/180405_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_DATA_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates <a href="analysis/180410_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_DATA_DeRubeis_ANNO_RR_estimate.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using WES informative prior, account for functional annotations in mutation rate adjustment <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_DeRubeis_ANNO_RR_estimate_account_for_functional_anno.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using LoF prior <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_DeRubeis_ANNO_RR_estimate_used_for_LoF_prior.html">link</a></p></li>
<li><p>Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model, more rate-adjusting variates, joint mutation rates adjustment model, intercepts and GC effect are study-specific, noXY chromosomes, no Krumm data when jointly adjusting mutation rates, using LoF prior, account for functional annotations in mutation rate adjustment <a href="analysis/180509_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_additional_covariates_joint_model_noXY_informative_prior_DATA_DeRubeis_ANNO_RR_estimate_account_for_functional_anno_using_LoF_prior.html">link</a></p></li>
</ul>
<p><strong>Some diagnostics of the problem in mutation rate adjustment</strong> 1. The ratio of syn. number and nonsyn. number in different studies.<a href="analysis/180413_diagnostics_of_the_number_of_syn_and_nonsyn_mutations_across_studies.html">link</a></p>
</div>
</div>
<div id="simulation-results" class="section level2">
<h2>1.3 simulation results</h2>
<ul>
<li><p>nonfunctioal synonymous mutations and LoF mutations (logRR as 0 and 3, respectively)<a href="analysis/180301_simulation_ANNO_RR_estimate.html">link</a>.</p></li>
<li><p>Mis3 mutations (logRR as 1)<a href="analysis/180301_simulation_ANNO_RR_estimate_Mis3.html">link</a>.</p></li>
</ul>
</div>
</div>
<div id="further-refining-mutation-rates" class="section level1">
<h1>2. Further refining mutation rates</h1>
<div id="investigation-of-ervs-derived-mutation-rates" class="section level2">
<h2>2.1 Investigation of ERVs derived mutation rates</h2>
<blockquote>
<p>Extremely rare variantsreveal patterns of germline mutation rate heterogeneity in humans</p>
</blockquote>
<ul>
<li>This paper uses 35 million extremely rare variants to estimate base-level mutation rates upon a 7-mer model with other genomic and epigenomic features.</li>
<li>The data can be downloaded <a href="http://mutation.sph.umich.edu/">here</a>, corresponding <a href="https://github.com/carjed/mr-eel">GitHub</a></li>
</ul>
</div>
<div id="compare-pair-likelihood-using-the-3-mer-mutation-rate-model-and-the-7-mer-mutation-rate-model" class="section level2">
<h2>2.1 compare pair likelihood using the 3-mer mutation rate model and the 7-mer mutation rate model</h2>
<blockquote>
<p>Leave out one chromosome during training and calculate the likelihood on the left-out chromosome.</p>
</blockquote>
<ul>
<li><p>Data analysis for WES data, using nonTADA genes to adjust for the effects of <code>div_score</code> and <code>GC_content</code><a href="./analysis/180321_compare_mutation_rate_models.html">here</a>.</p></li>
<li><p>Data analysis for WES data, using nonfunctional synonymous mutatoins to adjust for the effects of <code>div_score</code> and <code>GC_content</code><a href="./analysis/180417_compare_mutation_rate_models.html">here</a>.</p></li>
<li><p>Data analysis for WES data, using nonfunctional synonymous mutatoins, and addtional covaraites from the ERV model to adjust for the effects of <code>div_score</code> and <code>GC_content</code>, <code>RepT</code>, , <code>RecomboRate</code>, and several other epigenomic annotations<a href="./analysis/180418_compare_mutation_rate_models_with_additional_covariates.html">here</a>.</p></li>
<li><p>Data analysis for simulated data, data is simulated with the 7-mer ERV model, and fitted with both ERV model and the 3-mer Samocha model separately. Prediction is evaluated at the window level. Consider using other covariates, although the data is simulated without introducing effects from these covariates. <a href="./analysis/180419_simulate_mutations_and_fit_mutation_rate_model.html">here</a>.</p></li>
<li><p>Data analysis for simulated data, data is simulated with the 7-mer ERV model, and fitted with both ERV model and the 3-mer Samocha model separately. Prediction is evaluated at the base level. Consider using or not using other covariates, although the data is simulated without introducing effects from these covariates. <a href="./analysis/180426_mutation_rate_model_comparison_at_base_level.html">here</a>.</p></li>
<li><p>Data analysis for WES data, nonfunctional synonymous mutations are used to adust mutation rates. No covariates other than an intercept were used in fit the mutation rate model. <a href="./analysis/180501_prediction_accurary_of_mutation_model_between_ERV_and_3mer_for_WES_nonfunctional_syn_mutations_nocovariates.html">here</a>.</p></li>
</ul>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
