---
output:
  html_document:
    toc: true
---

<h1 style="color:grey;font-family:arial; position:relative; top:15px;text-align:center;" >TADA-A-plus<img src="website_materials/research_website_logo.png"  style="float:right;width:200px;height:150px;"></h1>
<br>

***



<br>

# 1. Analyze various coding annotations

<br>

> Annotations include 

+ Ribosnitch (allele-specific)
+ RBP binding sites
+ conservation (GERP)
+ Polyphen-damaging (Mis3)
+ Loss-of-function and Gain-of-function
+ CADD 

## 1.1 For WGS data

Use manuscript data/MSSNG/Simons_519, scaling factors are computed using WGS data, based on coding regions and regions 10kb within TSSs.


### 1.1.1 manuscript meta-analysis data
+ coding RR estimation.[link](analysis/180214_results_summary_manuscript_data_RR_estimation_with_mutrate_adjusted_for_div_coding_features.html)


### 1.1.2 MSSNG data
+ coding RR estimation (scaling file doesn't account for div score).[link](analysis/180216_RR_estimate_DATA_Yuen_2017_ANNO_RR_estimate.html)

+ coding RR estimation (scaling file accounting for div score).[link](analysis/180216_with_div_adjusted_RR_estimate_DATA_Yuen_2017_ANNO_RR_estimate.html)
* [Adjusting mutation rate accounting for div score](analysis/180215_recalibrate_mutation_rate_with_div_score_for_MSSNG_and_Simons_519.html)

+ About 75% of the Yuen et al. data used in the TADA-A manuscript are also found in the denovo-db Yuen data. 

Execution code to do data partition and reading-in is `code/pipeline00_generate_R_code_for_estimating_RR_using_partitioned_data`

### 1.1.3 Simons data

+ coding RR estimation (scaling file accounting for div score).[link](analysis/180216_with_div_adjusted_RR_estimate_Simons_519_new_version_ANNO_RR_estimate.html)
* [Adjusting mutation rate accounting for div score](analysis/180215_recalibrate_mutation_rate_with_div_score_for_MSSNG_and_Simons_519.html)

### 1.1.4 Wong NC et al. control data
+ coding RR estimation (scaling factors were calculated accouting for div score, using coding sequences and promoter-proximal non-coding sequences, used in the manuscript). [link](analysis/180221_with_div_adjusted_RR_estimate_Wong_control_ANNO_RR_estimate.html)

### 1.1.5 Neitherland 258 control

+ coding RR estimation (scaling factors were calculated accouting for div score, using coding sequences and promoter-proximal non-coding sequences, used in the manuscript[mutation rate adjustment](analysis/18308_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes.html))

+ [coding RR estimation accouting for div score](analysis/180221_with_div_adjusted_RR_estimate_Neitherland_control_ANNO_RR_estimate.html)

+ coding RR estimation accouting for div score, using the 7-mer ERV mutation rate model [link](analysis/180315_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Neitherland_258_control_ANNO_RR_estimate.html).

### 1.1.5 Iceland 1549 control control

+ coding RR estimation (scaling factors were calculated accouting for div score, using coding sequences and promoter-proximal non-coding sequences, used in the manuscript[mutation rate adjustment](analysis/18308_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes.html))

+ [coding RR estimation accouting for div score](analysis/180221_with_div_adjusted_RR_estimate_Iceland_control_ANNO_RR_estimate.html)

+ coding RR estimation accouting for div score, using the 7-mer ERV mutation rate model [link](analysis/180315_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iceland_1594_control_ANNO_RR_estimate.html).

## 1.2 For WES data
From denovo-db

| Name | PMID | Probands | Controls | Type |
|----|----|----|----|----|
|Iossifov | 25363768 | 2508 | 1911 | exome |
|Krumm | 25961944 | 2377 | 1786 | exome |
|DeRubeis2014 | 25363760 | 2270 | 0 | exome |

+ Need to use the nonfunctional syn mutations to adjust mutation rates.

+ When adjusting mutation rates, need to generate synonymous mutation rates for each 50 bp window, the file is here `../data/Example_windows_coding_with_nonfunctional_syn_mutrate.bed`

* [clean dataset](analysis/180215_invesigate_ASD_DNM_in_denovo_db_for_WES_studies.html)

+ [Adjust mutation rate using nonfunctional synonymous mutations](analysis/180215_recalibrate_mutation_rate_with_div_score_for_WES_data.html)

+ [Ajust mutation rate using all the mutations of nonTADA genes, which are genes with coding prior smaller than 0.05](analysis/18308_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes.html)

+ [Ajust mutation rate using all mutations of nonTADA genes, but using the 7-mer ERV muation rate model](analysis/180313_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonTADA_genes_ERV_mutrate.html)

+ [Ajust mutation rate using nonfunctional synonymous mutations, using the 7-mer ERV mutation rate](analysis/180315_recalibrate_mutation_rate_with_div_score_for_WES_data_using_nonfunctional_syn_ERV_mutrate.html)

### 1.2.1 Iossifov data
+ 2767 coding mutations after filtering, 490 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations.[link](analysis/180221_with_div_adjusted_RR_estimate_Iossifov_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in nonTADA genes.[link](analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Iossifov_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_ANNO_RR_estimate.html)

+ Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations.[link](analysis/180221_with_div_adjusted_RR_estimate_Iossifov_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonTADA genes.[link](analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Iossifov_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.[link](analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations, 7-mer ERV mutation rate model.[link](analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html)

### 1.2.2 Krumm data
+ 915 coding mutations after filtering, 119 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations.[link](analysis/180221_with_div_adjusted_RR_estimate_Krumm_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in nonTADA genes.[link](analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Krumm_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_ANNO_RR_estimate.html)

+ Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_ANNO_RR_estimate.html)


+ Use unaffected control data, Adjust mutation rate using nonfunctional synonymous mutations.[link](analysis/180221_with_div_adjusted_RR_estimate_Krumm_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using all mutations in nonTADA genes.[link](analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_Krumm_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.[link](analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Iossifov_control_ANNO_RR_estimate.html)

+ Use unaffected control data, Adjust mutation rate using nonTADA genes, 7-mer ERV mutation rate model.[link](analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_Krumm_control_ANNO_RR_estimate.html)


### 1.2.3 DeRubeis2014 data
+ 1590 coding mutations after filtering, 315 nonfunctional syn mutations, Adjust mutation rate using nonfunctional synonymous mutations. [link](analysis/180221_with_div_adjusted_RR_estimate_DeRubeis_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in using nonTADA genes. [link](analysis/180221_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_DeRubeis_ANNO_RR_estimate.html)

+ Adjust mutation rate using all mutations in nonTADA genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonTADA_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_DeRubeis_ANNO_RR_estimate.html)

+ Adjust mutation rate using nonfunctional synonymous mutations of all genes, 7-mer ERV mutation rate model. [link](analysis/180313_with_div_nonfunctional_syn_mutrate_calibration_adjusted_RR_estimate_ERV_mutrate_DATA_DeRubeis_ANNO_RR_estimate.html)

## 1.3 simulation results
+ nonfunctioal synonymous mutations and LoF mutations (logRR as 0 and 3, respectively)[link](analysis/180301_simulation_ANNO_RR_estimate.html).

+ Mis3 mutations (logRR as 1)[link](analysis/180301_simulation_ANNO_RR_estimate_Mis3.html).


# 2. Further refining mutation rates
## 2.1 Investigation of ERVs derived mutation rates
>Extremely rare variantsreveal patterns of germline mutation rate heterogeneity in humans

+ This paper uses 35 million extremely rare variants to estimate base-level mutation rates upon a 7-mer model with other genomic and epigenomic features.
+ The data can be downloaded [here](http://mutation.sph.umich.edu/), corresponding [GitHub](https://github.com/carjed/mr-eel)